filters/tee/tee_test.go

```diff
@@ -255,7 +256,9 @@ func TestTeeEndToEndBody2TeeRoutesAndClosing(t *testing.T) {
 
 	req.Host = "www.example.org"
 	req.Close = true
-	rsp, err := (&http.Client{}).Do(req)
+	rt := net.NewTransport(net.Options{})
+	defer rt.Close()
+	rsp, err := rt.RoundTrip(req)
```

Comment:
You should use proxy.Client() which is https://pkg.go.dev/net/http/httptest#Server.Client

Reply:
Why?

Reply:
I think I misunderstood the problem this is fixing.

> http client can sometimes create more than one request

Maybe its worth expanding when this can happen and how it is related to the flaky test this fixes.